controllers/ArticleController.js

```diff
@@ -438,4 +438,77 @@ export default class ArticleController {
       return next(err);
     }
   }
+
+  /**
+   * Report an article
+   *
+   * @static
+   *
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   *
+   * @returns {object} object with message converning reported article
+   *
+   * @memberof ArticleController
+   */
+  static async reportArticle(req, res, next) {
+    try {
+      const { slug } = req.params;
+      const userId = req.user.id;
+      const { reportTitle, reportBody } = req.body;
+
+      const article = await Article.findOne({ where: { slug } });
+      if (!article) return errorResponse(res, 404, { message: 'Article not found' });
+
+      const articleId = article.dataValues.id;
+
+      await Report.create({
+        userId,
+        articleId,
+        reportTitle,
+        reportBody,
+      });
+
+      const userReportsOnArticle = await Report.findAndCountAll({ where: { userId, articleId } });
+
+      // Multiple reports on an article by a user should count as 1 report
+      if (!(userReportsOnArticle.count > 1)) {
+        // Update reports count on article table
+        await Article.update(
+          { reports: (article.dataValues.reports + 1) },
+          { where: { slug } },
+        );
+      }
+
+      return successResponse(res, 201, 'message', 'Report submitted');
```

Comment:
Great work here Sam

I think this response could be a little confusing to track, wouldn't it be better if it has a key of `article` and an object with the `message`. This would be in sync with our other responses.

Reply:
Noted.. Thanks!